menu.py

```diff
@@ -102,100 +109,100 @@ def setup_menu(self, surface, background_colour=BLUE):
             in enumerate(menu_config)
         ]
 
-    def make_button(self, index, text, xpo, ypo, height, width, colour):
-        """Make a text button at the specified x, y coordinates
-        with the specified colour. Also adds a border (not configurable)"""
+    def make_button(self, index, text, xpo, ypo, colour, text_colour):
+        """Make a text button at the specified x, y coordinates"""
         logger.debug("Making button with text '%s' at (%d, %d)", text, xpo, ypo)
-        font = pygame.font.Font(None, 24)
-        label = font.render(str(text), 1, (colour))
-        self.screen.blit(label, (xpo, ypo))
         return dict(
             index=index,
             label=text,
-            rect=pygame.draw.rect(self.screen, CREAM, (xpo - 5, ypo - 5, width, height), 1)
+            btn = MyButton(label=text, pos=(xpo, ypo), col=colour, label_col=text_colour)
         )
 
-    def on_click(self, mousepos):
-        """Click handling function to check mouse location"""
-        logger.debug("on_click: %s", mousepos)
-        # Iterate through our list of buttons and get the first one
-        # whose rect returns True for pygame.Rect.collidepoint()
-        try:
-            button = next(obj for obj in self.buttons if obj['rect'].collidepoint(mousepos))
-            logger.info(
-                "%s clicked - launching %d",
-                button["label"], button["index"]
-            )
-            # Call button_handler with the matched button's index value
-            new_challenge = self.button_handler(button['index'])
-            return new_challenge
-        except StopIteration:
-            logger.debug(
-                "Click at pos %s did not interact with any button",
-                mousepos
-            )
-            return None
-
-    def button_handler(self, number):
+    def button_handler(self, event):
         """Button action handler. Currently differentiates between
         exit, rc and other buttons only"""
-        logger.debug("button %d pressed", number)
-        time.sleep(0.01)
-        if number < 7:
-            logger.info("other selected")
-            return "Other"
-        elif number == 7:
+        logger.debug("%s button pressed", event.label)
+        if event.label is "RC":
             logger.info("launching RC challenge")
-            new_challenge = RC(timeout=self.timeout)
+            new_challenge = RC(timeout=self.timeout, joystick=self.joystick)
             return new_challenge
-        elif number == 8:
+        elif event.label is "Exit":
             logger.info("Exit button pressed. Exiting now.")
             return "Exit"
         else:
-            return None
+            logger.info("other selected")
+            return "Other"
 
+    def joystick_handler(self, button):
+       if button['dright']:
```

Comment:
all of these `if` statements are exlcusive, correct? So let's `elif` the others as they don't need to be evaluated.

```
if thing:
    do_stuff
elif other_thing:
    do_other_stuff
```

Reply:
good catch, I'd left it like that from the first test code